### ISSUE TITLE
Invalid Client Error

### ISSUE THREAD

**Issue by azimhossaintuhin** [wagnerdelima/drf-social-oauth2#213]
After Creating An Application With Client Type "Confidential" and Authorization grant type:  Resource owner password-based
 When I Request An  Token 
And Hit The Route http://127.0.0.1:8000/auth/token
with Json Body This 
{
    "username": "username" ,
    "password":"password",
    "grant_type": "password",
    "client_id": "kRMUn5rVjGowZpqe8cqi20Ldghny5ARm7FLUke9g",
    "client_secret": "pbkdf2_sha256$720000$tNZxT5vI4mTyYgfrcZIZAO$MYQNjZCcURvHG18YsGfH4hSHrnjpXNcomoRYfXppVSc="
}
which is a valid client id and secret key it returns me  
{
    "error": "invalid_client"
}
but when i make the  Client Type "Public " and hit with the same data the access and refresh token will come

**Comment by bojanKuzma** [comment]
You must use the client secret before it is hashed by pbkdf2, you are using the hashed version. It says you to save the secret before saving the application.

**Comment by wagnerdelima** [comment]
That is correct. When you create your application, copy the client_secret (you can deselected the Hash checkbox) and save it. This shall solve your issue.

**Comment by antonioarmada** [comment]
I was facing the same issue, but I found that the hashed 'client_secret' isn't the cause of this error. It seem to be related to the django-oauth-toolkit module, if a use the old version 1.5.0, the 'invalid_client' error disappears. This old version works fine alongside the other updated modules. Here his is my working requirements.txt

Django==5.0.7
djangorestframework==3.15.2 
django-oauth-toolkit==1.5.0 # this is the version that works!
drf-social-oauth2==3.0.1
MarkupSafe==2.1.5 
oauthlib==3.2.2 
pyasn1==0.6.0 
pycparser==2.22
PyJWT==2.8.0
python-jose==3.3.0
pytz==2024.1 
requests==2.32.3 
requests-oauthlib==2.0.0 
rsa==4.9 
six==1.16.0 
social-auth-app-django==5.4.2 
social-auth-core==4.5.4 
setuptools==70.3.0

Something off-topic: the client_secret in the request is still necessary, but in the docs say that its'n ¿?

Thank for you work!

**Comment by bojanKuzma** [comment]
> I was facing the same issue, but I found that the hashed 'client_secret' isn't the cause of this error. It seem to be related to the django-oauth-toolkit module, if I use the old version 1.5.0, the 'invalid_client' error disappears. This old version works fine alongside the other updated modules. Here is my working requirements.txt
> 
> Django==5.0.7 djangorestframework==3.15.2 django-oauth-toolkit==1.5.0 # this is the version that works! drf-social-oauth2==3.0.1 MarkupSafe==2.1.5 oauthlib==3.2.2 pyasn1==0.6.0 pycparser==2.22 PyJWT==2.8.0 python-jose==3.3.0 pytz==2024.1 requests==2.32.3 requests-oauthlib==2.0.0 rsa==4.9 six==1.16.0 social-auth-app-django==5.4.2 social-auth-core==4.5.4 setuptools==70.3.0
> 
> Something off-topic: the client_secret in the request is still necessary, but in the docs say that its'n ¿?
> 
> Thank for your work!

For this reported issue it was the hashed client secret, in the django-oauth-toolkit [documentation](https://django-oauth-toolkit.readthedocs.io/en/latest/tutorial/tutorial_01.html#test-your-authorization-server) it is also mentioned that the client secret "must be before hashing, should not begin with ‘pbkdf2_sha256’ or similar". Maybe you can provide more information like a curl request that you sent?

**Comment by antonioarmada** [comment]
es, but in this case the error was caused by something related to the version of the module.  As I was testing, I used a client_secret='client_secret', which of course encrypted after saving.  With these credentials I got the 'invalid_client' error, but without changing anything other than the version of the module the error disappeared and I got the desired token :)

curl -X POST -d "grant_type=convert_token&client_id=client_id&client_secret=client_secret&backend=google-oauth2&token=ya29.a0AXoo......" http://localhost:8000/auth/convert-token

with the updated version of django-oauth-toolkit, I got:

{
  "error": "invalid_client"
}

same curl.. but with the django-oauth-toolkit==1.5.0 version module I finally got:

{
  "access_token": "eyJhbGciOiJIUzI1N.....",
  "expires_in": 36000,
  "token_type": "Bearer",
  "scope": "read write",
  "refresh_token": "eyJhbGciOiJIUzI1NiI....."
}

**Comment by wagnerdelima** [comment]
Thank you the thread discussion. I did see the client secret should be copied before hashing. So yeah, the old version of django-oauth-toolkit doesn't include that update. That is why it works.

The newest version of drf-social-oauth2 doesn't require the client secret as it may represent a risk if exposed. So, in your requests, just remove the client secret from the list of params. 

Closing this for now. Thanks for engaging.
